azext_edge/edge/providers/check/cloud_connectors.py

```diff
@@ -56,6 +58,38 @@ def process_cloud_connector(
         "items", []
     )
 
+    if connector_resource_name:
+        filtered_connectors = filter_resources_by_name(
+            resources=all_connectors, resource_name=connector_resource_name
+        )
+
+        excluded_connectors = [
+            connector
+            for connector in all_connectors
+            if connector not in filtered_connectors
+        ]
+
+        all_connectors = filtered_connectors
+
+        # get (name, namespace) tuples for excluded connectors
+        excluded_connector_properties = [
+            (
+                get_resource_metadata_property(connector, prop_name="name"),
+                get_resource_metadata_property(connector, prop_name="namespace")
+            ) for connector in excluded_connectors
+        ]
+
+        # filter out topic maps with both excluded connector name and namespace
+        all_topic_maps = [
+            map
+            for map in all_topic_maps
+            if (
+                map.get("spec", {}).get(topic_map_reference_key),
+                get_resource_metadata_property(map, prop_name="namespace"),
+            )
+            not in excluded_connector_properties
+        ]
```

Comment:
Can we add some unit tests specifically for this chunk of logic?
I followed through with some sample data and it looks like it's including / excluding the right things, but would be great to have a test where we can throw a bunch of connector names, namespaces, and topic maps into a function and make sure we get the right ones out at the other end.

Reply:
thanks for testing! I will update with unit test

Reply:
Hi Ryan, in order to test this logic I put them in a new function, since eval_function only return things in display manager and does not have detail of topic maps, please let me know if there's more concern to the tests!